### PR TITLE
S2U-9 5.1.1.5 Tests & Quizzes: PDF export for assessment submissions …

### PR DESCRIPTION
**Pull request by JuanDavid102**
…- fix

### CODE REVIEW

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/evaluation/ExportAction.java [hunk 1]
```diff
@@ -116,7 +116,7 @@ public void processAction(ActionEvent ae) {
 			double currentScore = deliveryBean.getTableOfContents().getCurrentScore();
 			double maxScore = deliveryBean.getTableOfContents().getMaxScore();
 			DecimalFormat twoDecimalsFormat = new DecimalFormat("0.00");
-			this.addCellToTable(shortSummaryTable, rb.getFormattedMessage("score_format", new String[]{twoDecimalsFormat.format(currentScore), twoDecimalsFormat.format(maxScore), twoDecimalsFormat.format((currentScore / maxScore) * 100)}), 0, 0);
+			this.addCellToTable(shortSummaryTable, rb.getFormattedMessage("score_format", new String[]{twoDecimalsFormat.format(currentScore), twoDecimalsFormat.format(maxScore), (maxScore==0)? "0" : twoDecimalsFormat.format((currentScore / maxScore) * 100)}), 0, 0);
```
Comment (mpellicer) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/evaluation/ExportAction.java:119]: Extract this into a new line and put some spaces after the '?'

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/evaluation/ExportAction.java [hunk 2]
```diff
@@ -542,37 +542,71 @@ public PdfPTable getQuestionTitle(String questionText, boolean showAllInformatio
...
+		table.addCell(latexCell);
+	}
+
+	/**
+	 * Method to create a Paragraph with Latex functions.
+	 * 
+	 * @param text
+	 * @return Paragraph latexParagraph
+	 */
+	public Paragraph createLatexParagraph(String text) {
```
Comment (mpellicer) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/evaluation/ExportAction.java:557]: This has to be private

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/evaluation/ExportAction.java [hunk 3]
```diff
@@ -542,37 +542,71 @@ public PdfPTable getQuestionTitle(String questionText, boolean showAllInformatio
 	 * @param table - PdfPTable where save the resolved text
 	 */
 	public void addLatexFunctionsToTable(String text, PdfPTable table) {
```
Comment (mpellicer) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/evaluation/ExportAction.java:544]: MAke it private

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/evaluation/ExportAction.java [hunk 4]
```diff
@@ -542,37 +542,71 @@ public PdfPTable getQuestionTitle(String questionText, boolean showAllInformatio
...
+				float finalHeight = formula.createBufferedImage(TeXFormula.BOLD, 10, null, null).getHeight(null);
+				pdfLatexImage.scaleAbsolute(finalWidth, finalHeight);
+
+				latexParagraph.add(new Chunk(textBeforeLatex));
+				latexParagraph.add(new Chunk(pdfLatexImage, -1, -2, true));
+
+				currentSearch = 1;
+				if (text.indexOf(searchIndex[0], latexFinalIndex + 2) != -1) {
+					currentSearch = 0;
+					if (text.indexOf(searchIndex[1], latexFinalIndex + 2) != -1){
```
Comment (mpellicer) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/evaluation/ExportAction.java:593]: Spacing

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/evaluation/ExportAction.java [hunk 5]
```diff
@@ -542,37 +542,71 @@ public PdfPTable getQuestionTitle(String questionText, boolean showAllInformatio
...
+				}
 
-			latexInitIndex = text.indexOf("$$", latexFinalIndex + 2);
-			if (latexInitIndex != -1) {
-				textBeforeLatex = text.substring(latexFinalIndex, latexInitIndex).replace("$$", "");
-				latexFinalIndex = text.indexOf("$$", latexInitIndex + 2);
+				latexInitIndex = text.indexOf(searchIndex[currentSearch], latexFinalIndex + 2);
+				
+				if (latexInitIndex != -1) {
+					textBeforeLatex = text.substring(latexFinalIndex, latexInitIndex).replace("$$", "").replace("\\(\\", "").replace("\\)", "");
```
Comment (mpellicer) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/evaluation/ExportAction.java:601]: Consider defining separators as constants and reuse them.

Comment (mpellicer) [reply samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/evaluation/ExportAction.java:601]: LATEX_DOLLAR_SEPARATOR

**Review by mpellicer — CHANGES_REQUESTED** [review]
Requested minor changes and improvements, thanks for the very good work.

**Comment by mpellicer** [comment]
@JuanDavid102 Tell me when you are ready please!

**Comment by JuanDavid102** [comment]
Yes, I finished it

**Review by mpellicer — APPROVED** [review]

**Pull request merged**